### PR TITLE
Publish to Apache nightlies server

### PR DESCRIPTION
**Pull request by seglo**
Snapshots are published to https://nightlies.apache.org/pekko/snapshots/ on a daily cron schedule. Only the last 30 snapshots are kept.

### CODE REVIEW

File: .github/workflows/publish-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Sync from remote
+        uses: ./.github/actions/sync-nightlies
+        with:
+          switches: -avzh --update --delete --progress
+          local_path: repo
+          remote_path: ${{ secrets.NIGHTLIES_RSYNC_PATH }}/pekko/snapshots
```
Comment (pjfanning) [inline .github/workflows/publish-nightly.yml:76]: @seglo is the apache github organization setup with these secrets?

Comment (seglo) [reply .github/workflows/publish-nightly.yml:76]: @pjfanning Yes it should be. I have an open ticket with INFRA. https://issues.apache.org/jira/browse/INFRA-23895

**Review by pjfanning — APPROVED** [review]
lgtm - this is really useful

**Comment by nvollmar** [comment]
Great work, now we can continue working on surrounding projects, like switching packages to pekko  👍

**Pull request merged**